File: Workshops/WapMass-Workshop/Android/Volley/app/src/main/java/bawo/volley/MainActivity.java
package bawo.volley;


import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;


import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.VolleyLog;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Arrays;

public class MainActivity extends AppCompatActivity {

    private final static String URL_String = "http://magadistudio.com/complete-android-developer-course-source-files/string.html";
    private final static String JSON_ARRAY_URL = "https://hacklab2018-14718.firebaseio.com/hospitals.json";
    private final static  String JSON_OBJ_URL = "https://earthquake.usgs.gov/fdsnws/event/1/query?format=geojson&starttime=2014-01-01&endtime=2014-01-02";

    private RequestQueue queue;
    private ArrayList data;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //RequestQueue queue = Volley.newRequestQueue(this);
        queue = Volley.newRequestQueue(this);
        data = new ArrayList();
        getFacilities(JSON_ARRAY_URL);


    }


    //Get Movies
    private void getFacilities(String url) {
        JsonArrayRequest facilities = new JsonArrayRequest(Request.Method.GET,
                url, new Response.Listener<JSONArray>() {
            @Override
            public void onResponse(JSONArray response) {
                for(int i=0; i< 2; i++){
                    try {
                        JSONObject  facilityObject = response.getJSONObject(i);
                        data.add(facilityObject);
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                }
            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });
        queue.add(facilities);
    }
    private void getJsonObject(String url) {
        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET,
                url, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {

                try {


                    // Log.d("Object: ", response.getString("type").toString());

                    JSONObject metadata = response.getJSONObject("metadata");

                    Log.d("Metadata: ", metadata.toString());
                    Log.d("MetaInfo: ", metadata.getString("generated"));
                    Log.d("MetaInfo: ", metadata.getString("url"));
                    Log.d("MetaInfo: ", metadata.getString("title"));


                    /// jsonArray
                    JSONArray features = response.getJSONArray("features");

                    for (int i = 0; i < features.length(); i++) {

                        //Get objects
                        JSONObject propertiesObj = features.getJSONObject(i).getJSONObject("properties");

                        Log.d("Place:", propertiesObj.getString("place"));


                    }






                } catch (JSONException e) {
                    e.printStackTrace();
                }


            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });

        queue.add(jsonObjectRequest);


    }


    public void getStringObject(String url) {

        StringRequest stringRequest = new StringRequest(Request.Method.GET, url, new Response.Listener<String>() {
            @Override
            public void onResponse(String response) {

                Log.d("My String: ", response.toString());

            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });

        queue.add(stringRequest);

    }


}

File: KotlinForAndroid/android/VolleyString/app/src/main/java/choreappappkotlinandroid/gohool/com/volleystring/MainActivity.kt
package choreappappkotlinandroid.gohool.com.volleystring

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.Response
import com.android.volley.VolleyError
import com.android.volley.toolbox.JsonArrayRequest
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    var volleyRequest: RequestQueue? = null
    val earthquakeLink = "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/1.0_hour.geojson"
    val movieLink = "http://netflixroulette.net/api/api.php?director=Quentin%20Tarantino"
    val stringLink = "https://www.magadistudio.com" +
            "/complete-android-developer-course-source-files/string.html"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        volleyRequest = Volley.newRequestQueue(this)
       // getString(stringLink)
        //getJsonArray(movieLink)
        getJsonObject(earthquakeLink)


    }

    fun getJsonObject(Url: String) {
        val jsonObject = JsonObjectRequest(Request.Method.GET, Url,
                Response.Listener {
                    response: JSONObject ->
                    try {

                        val metadataObject = response.getJSONObject("metadata")
                        Log.d("Metadata:", metadataObject.getString("title"))


                        //Get the features array
                        val featuresArray =
                                response.getJSONArray("features")
                        for (i in 0.. featuresArray.length() - 1) {
                            //Get the object within
                            val propertyObj = featuresArray.getJSONObject(i)
                                    .getJSONObject("properties")

                            Log.d("Properties Place: ", propertyObj.getString("place"))
                        }




                    }catch (e:JSONException) {e.printStackTrace()}
                },
                Response.ErrorListener {  })

        volleyRequest!!.add(jsonObject)
    }

    fun getJsonArray(Url: String){
        val jsonArray = JsonArrayRequest(Request.Method.GET, Url,
                Response.Listener {
                    response: JSONArray -> try {

                    Log.d("Response", response.toString())

                    for (i in 0..response.length() - 1) {

                        val movieObj = response.getJSONObject(i)

                        var movieName = movieObj.get("show_title")
                        Log.d("MovieName:", movieName.toString())
                    }


                    } catch (e: JSONException){ e.printStackTrace()}
                },
                Response.ErrorListener {
                    error: VolleyError? ->
                      try {
                          Log.d("Error", error.toString())

                      }catch (e: JSONException) {
                          e.printStackTrace()
                      }
                })

        volleyRequest!!.add(jsonArray)
    }

    fun getString(Url: String){

        val stringReq = StringRequest(Request.Method.GET, Url,
                Response.Listener<String> {
                    response: String? ->
                      try {
                          Log.d("Stuff", response.toString())

                      }catch (e: JSONException){
                          e.printStackTrace()
                      }

                }, Response.ErrorListener {
                   error: VolleyError? ->
                        try {
                            Log.d("Error:", error.toString())

                        }catch (e: JSONException) {
                            e.printStackTrace()
                        }



        })

        volleyRequest!!.add(stringReq)


    }


}
